tests/mechanic/telemetry_test.py

```diff
@@ -2149,22 +2148,25 @@ class DiskIoTests(TestCase):
     @mock.patch("esrally.metrics.EsMetricsStore.put_count_node_level")
     def test_diskio_process_io_counters(self, metrics_store_node_count, process_io_counters):
         Diskio = namedtuple("Diskio", "read_bytes write_bytes")
-        process_start = Diskio(10,10)
-        process_stop = Diskio(11,11)
+        process_start = Diskio(10, 10)
+        process_stop = Diskio(11, 11)
         process_io_counters.side_effect = [process_start, process_stop]
 
         tmp_dir = tempfile.mkdtemp()
         cfg = create_config()
         metrics_store = metrics.EsMetricsStore(cfg)
         
-        device = telemetry.DiskIo(metrics_store, 1, tmp_dir, "rally0")
+        device = telemetry.DiskIo(metrics_store, node_count_on_host=1, log_root=tmp_dir, node_name="rally0")
         t = telemetry.Telemetry(enabled_devices=[], devices=[device])
         node = cluster.Node(pid=None, host_name="localhost", node_name="rally0", telemetry=t)
         t.attach_to_node(node)
         t.on_benchmark_start()
-        t.on_benchmark_stop()
         t.detach_from_node(node, running=True)
         t.detach_from_node(node, running=False)
+        node2 = cluster.Node(pid=None, host_name="localhost", node_name="rally0", telemetry=t)
```

Comment:
What I meant in https://github.com/elastic/rally/pull/721#discussion_r303944880 is that we recreate the telemetry device, not the node. This ensures that we don't rely on any state that is tied to that instance.